### DISCUSSION TITLE
onActivitybarClicked method ?

### DISCUSSION THREAD

**Discussion by G3orge26** [microsoft/vscode-discussions#313] [Extension Development QnA] [answered]
Hi all,

I was wondering if its is possible to have an Activitybar that:

- Has an onClick attached to it to run some code when someone clicks on it
- Doesnt necessarily require a sidebar attached to it

Alternatively, can I somehow grab that UI element and attach a callback on it myself ?

I have been scavenging the documentation but I cant find any information on that

Thank you in advance

**Comment by G3orge26** [comment]
As some secondary information, if I add an activitybar element on the contributes section with a title, name, and icon but without a views element for the activitybar ID, then the icon wont show on the side aafter the 5 original vscode Icons/buttons

**Comment by isidorn** [comment, marked as answer]
This is currently not possible through the VS Code API. More about the API can be found here https://code.visualstudio.com/api
